rclcpp_action/src/server.cpp

```diff
@@ -483,6 +491,15 @@ ServerBase::execute_cancel_request_received(std::shared_ptr<void> & data)
       pimpl_->action_server_.get(), &request_header, response.get());
   }
 
+  if (ret == RCL_RET_TIMEOUT) {
+    GoalUUID uuid = request->goal_info.goal_id.uuid;
+    RCLCPP_WARN(
+      pimpl_->logger_,
+      "Failed to send cancel response %s (timeout): %s",
+      to_string(uuid).c_str(), rcl_get_error_string().str);
+    rcl_reset_error();
+    return;
```

Comment:
Wouldn't it make sense to change the method signature to something else than void ?

Reply:
hmm, probably. so that ServerBase can what needs to be done based on the return code. i am not gonna do that in this PR, but that can be follow-up issue. (i am considering to backport this iron and humble.) thanks!